public/app/plugins/panel/canvas/components/connections/ConnectionAnchors2.tsx

```diff
@@ -0,0 +1,171 @@
+import { css } from '@emotion/css';
+import { useRef } from 'react';
+import * as React from 'react';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { useStyles2 } from '@grafana/ui';
+import { ConnectionCoordinates } from 'app/features/canvas/element';
+
+type Props = {
+  setRef: (anchorElement: HTMLDivElement) => void;
+  setAnchorsRef: (anchorsElement: HTMLDivElement) => void;
+  handleMouseLeave: (
+    event: React.MouseEvent<Element, MouseEvent> | React.FocusEvent<HTMLDivElement, Element>
+  ) => boolean;
+};
+
+export const CONNECTION_ANCHOR_DIV_ID = 'connectionControl';
+export const CONNECTION_ANCHOR_ALT = 'connection anchor';
+export const CONNECTION_ANCHOR_HIGHLIGHT_OFFSET = 8;
+// Unit is percentage from the middle of the element
+// 0, 0 middle; -1, -1 bottom left; 1, 1 top right
+export const ANCHORS = [
+  { x: -1, y: 1 },
+  { x: -0.5, y: 1 },
+  { x: 0, y: 1 },
+  { x: 0.5, y: 1 },
+  { x: 1, y: 1 },
+  { x: 1, y: 0.5 },
+  { x: 1, y: 0 },
+  { x: 1, y: -0.5 },
+  { x: 1, y: -1 },
+  { x: 0.5, y: -1 },
+  { x: 0, y: -1 },
+  { x: -0.5, y: -1 },
+  { x: -1, y: -1 },
+  { x: -1, y: -0.5 },
+  { x: -1, y: 0 },
+  { x: -1, y: 0.5 },
+];
+export const ANCHOR_PADDING = 3;
+export const HALF_SIZE = 2.5;
+
+const zIndex = {
```

Comment:
I saw that sneaky `999` somewhere else in the code... seems like that sort of relates to these values, maybe it should also be managed here.